### ISSUE TITLE
[Bug] Auth Unity SDK still requires Google Play Services while Android SDK does not

### ISSUE THREAD

**Issue by MohsenneChaverdie** [firebase/quickstart-unity#975]
### Issue details:
  * Unity editor version: 2019.2.21f1
  * Firebase Unity SDK version: 7.1.0
  * Source you installed the SDK: unitypackage
  * Problematic Firebase Component: Auth
  * Other Firebase Components in use: Firestore
  * Platform you are using the Unity editor on: Windows
  * Platform you are targeting: Oculus Quest 2
  * Scripting Runtime: Mono 
 
### Issue description:
Trying to authenticate with Firebase through the Auth Unity package. In Unity Editor and Android with Google Services it works, however on Android devices without Google services, like Oculus Quest 2, or Emulator (No google API) does not work.
On pure Android, through Android Studio, the Authentication does not need Google play services. So why the Unity Auth Package still requires Google API?

#### Steps to reproduce:
**Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?** Yes. I have tried the Auth example.

**What's the issue repro rate?** 100%

**What happened? How can we make the problem occur?**
Try a simple Unity scene with authentication on an Android without Google play services. It can never authenticate itself.

**On which documentation page have you seen that Google Play Services for Authentication is not required anymore?**
https://firebase.google.com/docs/android/android-play-services

**Comment by chkuang-g** [comment]
@MohsenneChaverdie 

Could you provide the log when the device failed to authenticate? That would help us to quickly identify the issue.

I suspect it is because C++ SDK is still checking if Google Play Services is available while then Android SDK it wraps actually does not require it anymore. See [this code](https://github.com/firebase/firebase-cpp-sdk/blob/e86aa624a97ed172056fc76a4902ab6257f3f8bc/auth/src/auth.cc#L49).

If this is the case, you should see [FirebaseApp.CheckAndFixDependenciesAsync()](https://firebase.google.com/docs/reference/unity/class/firebase/firebase-app#checkandfixdependenciesasync) failed.

The Android public doc said that `com.google.firebase:firebase-auth:20.0.3` does not requires Google Play Service.  However, Firebase Unity SDK `7.1.0` is using `com.google.firebase:firebase-auth:20.0.2`.  I think this version probably does not require GPS anyway but I need to confirm with the team.

I think this is a valid bug.  Thanks for reporting it.

**Comment by MohsenneChaverdie** [comment]
@chkuang-g thank you very much for your reply. 

Yes exactly. I see the FirebaseApp.CheckAndFixDependenciesAsync() fails on Android without Google Play Services.

Someone on [Twitter ](https://twitter.com/pux0r3/status/1366921826828386306?s=20)apparently is very sure that at this time due to architectural problems, it is not possible.

**Comment by chkuang-g** [comment]
@MohsenneChaverdie Thanks for your response.  I'll contact the team about this.

**Comment by StephenHodgson** [comment]
I'm also blocked by this as well.

**Comment by StephenHodgson** [comment]
I'm also curious why all of the other Firebase libraries are open source on GitHub with the exception being for the Unity one. 

I'm also surprised you guys don't have a dedicated C# library that you just build the Unity packages off of

**Comment by wimachtendink** [comment]
So, forgive my cynicism but I think a lot of people know who Stephen Hodgson is in the XR world, and it also looks like you literally just had to remove two lines of code in 4  locations.

I understand you guys are busy but this feels a little like you don't care about issues until someone important mentions it  which is a pretty lame way to handle things.

I think you guys should feel a little bit of shame for how you have handled this.

**Comment by StephenHodgson** [comment]
@wimachtendink I am quite humbled about your remark of my value to the XR space, but it's only been a month since this issue was opened. Not too bad if these guys are running the project in their free time at work or if they have to dedicate a limited amount of hours to each project.

It could have been fixed a bit faster had this package been open source like the rest of the firebase libraries.

Having said that, there might be some generation for swig that happens be a manual process, but could also be automated if setup correctly. I image that was the only real reason why the library isn't in GitHub.

The real fix would be providing a strictly .NET/C# library for the firebase built on top of the https://github.com/firebase/firebase-cpp-sdk and then building the unity implementation off of that.

**Comment by StephenHodgson** [comment]
Oh wait I take it back. It's been 2 years lol
https://github.com/firebase/quickstart-unity/issues/223

**Comment by wimachtendink** [comment]
Maybe I was too harsh, but a year ago they basically told everyone "Shut up about supporting Non-play services!" in #223 and then about 6 months ago (maybe less I couldn't find the release notes) they finally removed the requirement. Then, once the requirement had been removed they just didn't fix it - because they were still telling everyone to shut up and write their own REST APIs #938 which was actually the same issue as this thread.

Anyway, Issues isn't the place for ranting like this - but it makes me feel a bit like making a new issue, something like "Devs dismissive of user issues"

But I'm glad it'll finally be getting fixed.

**Comment by vimanyu** [comment]
Thanks for your patience everyone. I am closing this ticket and this will go out in the next release. Please note that the fix is on the C++ side but we will have to wait until the next Unity release for the Unity plugins to support this.

**Comment by StephenHodgson** [comment]
@vimanyu do you have any ETA when the next release would go out? Do you guys have a regular cadence you follow?

TBH I usually only follow release notifications on GitHub, but this library isn't in GitHub like the rest of the firebase libraries.
